lib/OpenQA/WebAPI/Controller/Admin/Needle.pm

```diff
@@ -124,41 +124,70 @@ sub module {
 
 sub delete {
     my ($self) = @_;
-    my @removed_ids;
-    my @errors;
-    for my $needle_id (@{$self->every_param('id')}) {
-        my $needle = $self->db->resultset('Needles')->find($needle_id);
-        if (!$needle) {
-            push(
-                @errors,
-                {
-                    id           => $needle_id,
-                    display_name => $needle_id,
-                    message      => "Unable to find $needle_id"
-                });
-            next;
-        }
 
-        my $error = $needle->remove($self->current_user);
-        if ($error) {
-            push(
-                @errors,
-                {
-                    id           => $needle_id,
-                    display_name => $needle->filename,
-                    message      => $error
-                });
-        }
-        else {
-            push(@removed_ids, $needle_id);
-        }
+    # check whether Minion worker are available to get a nice error message instead of an inactive job
+    my $gru = $self->gru;
+    if (!$gru->has_workers) {
+        return $self->render(
+            json => {error => 'No Minion worker available. The <code>openqa-gru</code> service is likely not running.'}
+        );
```

Comment:
Shouldn't we assume that there is always a Gru/Minion worker?

Reply:
In theory yes. In practice Gru/Minion is realized as a separate service which might not be started. A lot of problems people come to me with are simply resolved by starting a missing service. This hopefully helps people to help themselves.